### PR TITLE
✨ Use kustomize new syntax for patches

### LINKED ISSUE #1730: templates: 'patchesStrategicMerge' is deprecated
/kind feature

**Describe the solution you'd like**

While running `make templates`, we can see the following output:
```
# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
```

Can be observed in the CI output too. 

**Anything else you would like to add:**

This comes from the new Kustomize version being used: https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.0.0
> We are deprecating the following kustomization fields. These fields will never be removed from the kustomize.config.k8s.io/v1beta1 Kustomization API, but they will not be included in the kustomize.config.k8s.io/v1 Kustomization API. When Kustomization v1 is available, we will announce the deprecation of the v1beta1 version. There will be at least two releases between deprecation and removal of Kustomization v1beta1 support from the kustomize CLI, and removal itself will happen in a future major version bump:
> * patchesStrategicMerge. Please use patches instead

### PR DESCRIPTION
**Pull request by mquhuy**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1730 

**Special notes for your reviewer**:
`Kustomize` version needs to be updated to 5.2.1, as the current version (5.1.1) gets confused with multiple objects in the same patch file (it only understands if the patch file is split so that each object is in one file).

I also updated the kustomize version in `ensure_kustomize.sh`. Not sure if it's in use though, as I can see that the version is already very old.

### CODE REVIEW

File: kustomize/v1alpha7/flatcar/kustomization.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 resources:
 - ../default
 
-patchesStrategicMerge:
-- patch-flatcar.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Comment (lentzi90) [inline kustomize/v1alpha7/flatcar/kustomization.yaml:5]: It is not technically anything wrong with having the `apiVersion` and `kind` in the middle of the file, but I would prefer that we keep it at the top (if specified at all). Now it looks a bit like it has something to do only with the `patches` :smile:

File: kustomize/v1alpha7/default/kustomization.yaml [hunk 2]
```diff
@@ -1,2 +1,4 @@
 resources:
 - cluster-template.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Comment (lentzi90) [inline kustomize/v1alpha7/default/kustomization.yaml:4]:
```suggestion
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- cluster-template.yaml
```

File: kustomize/v1alpha7/without-lb/kustomization.yaml [hunk 3]
```diff
@@ -1,5 +1,7 @@
 resources:
 - ../default
 
-patchesStrategicMerge:
-- patch-without-lb.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Comment (lentzi90) [inline kustomize/v1alpha7/without-lb/kustomization.yaml:5]: Same here

**Review by lentzi90 — COMMENTED** [review]
Looks good! Could you move the `apiVersion` and `kind` to the top of the files?

File: kustomize/v1alpha7/without-lb/kustomization.yaml (hunk 3 above)
Comment (mquhuy) [reply kustomize/v1alpha7/without-lb/kustomization.yaml:5]: Sure. I'll do that.

File: kustomize/v1alpha7/default/kustomization.yaml (hunk 2 above)
Comment (mquhuy) [reply kustomize/v1alpha7/default/kustomization.yaml]: Sure

**Review by lentzi90 — COMMENTED** [review]
Great!
I think we can also update the same in [v1alpha6](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/tree/main/kustomize/v1alpha6), [v1alpha5](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/tree/main/kustomize/v1alpha5) and [here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/test/e2e/data/kustomize/md-remediation/kustomization.yaml). Then I think we got rid of all the old syntax :slightly_smiling_face:

**Review by lentzi90 — COMMENTED** [review]
/approve

**Comment by lentzi90** [comment]
Triggering full e2e since this touches the remediation tests
/test pull-cluster-api-provider-openstack-e2e-full-test

**Comment by lentzi90** [comment]
It is unclear to me if the test failure is related to the PR
/test pull-cluster-api-provider-openstack-e2e-test

**Pull request merged**